README.rst
.. This README is meant for consumption by humans and pypi. Pypi can render rst files so please do not use Sphinx features.
   If you want to learn more about writing documentation, please check out: http://docs.plone.org/about/documentation_styleguide.html
   This text does not appear on pypi or github. It is a comment.

==============================================================================
kitconcept.voltodemo
==============================================================================

.. image:: https://kitconcept.com/logo.svg
   :alt: kitconcept
   :target: https://kitconcept.com/


.. image:: https://secure.travis-ci.org/collective/kitconcept.voltodemo.png
    :target: http://travis-ci.org/collective/kitconcept.voltodemo

kitconcept.voltodemo is a helper package to setup a Plone site ready to use
with Volto. Drop it in your buildout and then install it. It is used in Volto
itself for testing it.


Usage
-----

https://github.com/plone/volto/blob/master/api/base.cfg#L13

and along with plonesite recipe:

https://github.com/plone/volto/blob/master/api/base.cfg#L13

Demo home page
--------------

It features a hack to make the Plone site Volto Blocks enabled with some demo
content.

Volto Editor
-------------

It enables the Volto Blocks behavior on the ``Document`` content type by
default, enabling Volto editor for that content type.

Document content type
---------------------

``Richtext`` and ``table of contents`` behaviors has been removed from the ``Document`` behaviors since it's confusing for the users if it shows in the form.

Disabled content types
----------------------

It disables (fti) of some default content types since they are not ready or
full working yet on Volto side. So, ``Collections``, ``Link`` and ``Events``
are disabled.

CORS profile
------------

A quick helper for enable CORS for development config is also provided in the
``kitconcept.voltodemo`` module. So you can call::

  <include package="kitconcept.voltodemo.cors" />

from you ZCML while developing.

Versions compatibility
----------------------

kitconcept.voltodemo < 2 Volto 3 and Volto 4 until alpha 9
kitconcept.voltodemo >= 2 Volto 4 (from alpha 10)


CHANGES.rst
Changelog
=========


2.3 (unreleased)
----------------

- Nothing changed yet.


2.2 (2019-12-03)
----------------

- Add Volto blocks :)
  [sneridagh]


2.1 (2019-12-03)
----------------

- Removed Richtext and toc behaviors from Document content type
  [sneridagh]


2.0 (2019-10-31)
----------------

- Patch Password reset tool in Products.CMFPlone to use the optional volto_domain in the
  e-email which is sent to users, only if the request is made through REST.
  [fredvd]

- Add Volto settings control panel with frontend_domain field.
  [fredvd]

- Renamed from tiles -> blocks
  [sneridagh]


1.4 (2019-09-12)
----------------

- Re-add the link to be available.
  [sneridagh]


1.3 (2019-08-30)
----------------

- Re-add the collection to be available.
  [sneridagh]


1.2.0 (2019-08-30)
------------------

- UnPin p.restapi, because of conflicts, meh
  [sneridagh]


1.1.3 (2019-08-30)
------------------

- Pin p.restapi version.
  [sneridagh]


1.1.2 (2019-08-28)
------------------

- Update the text with more agnostic numbers.
  [sneridagh]


1.1.1 (2019-06-09)
------------------

- Update version numbers
  [sneridagh]


1.1.0 (2019-05-04)
------------------

- Documentation.
  [sneridagh]

- Clean up package, Python 3, tests.
  [sneridagh]


1.0.1 (2019-04-18)
------------------

- Fix small typos and link ranges
  [sneridagh]


1.0.0 (2019-04-18)
------------------

- Amend the text on the home page.
  [sneridagh]

- Releasing 1.0.0
  [sneridagh]


1.0.0a17 (2019-04-16)
---------------------

- Add collective.folderishtypes
  [sneridagh]


1.0.0a16 (2019-04-16)
---------------------

- Renamed optional cors configuration module, install restapi from scratch.
  [sneridagh]


1.0.0a15 (2019-04-15)
---------------------

- Add optional module with CORS ZCML.
  [sneridagh]


1.0.0a14 (2019-04-15)
---------------------

- Update to Python3 and Plone 5.2
  [sneridagh]


1.0.0a13 (2019-03-25)
---------------------

- Update versions on homepage.
  [sneridagh]


1.0.0a12 (2019-03-15)
---------------------

- Update versions on homepage.
  [sneridagh]


1.0.0a11 (2019-03-01)
---------------------

- Remove PAS plugin hack since in Volto 1.6.0 it is not required anymore.
  [sneridagh]


1.0.0a10 (2019-02-20)
---------------------

- Add a *inner* admin user to overcome the limitation on using Zope users.
  [sneridagh]


1.0.0a9 (2019-02-19)
--------------------

- Adding the PAS plugin to workaround the images/files pull from HTML.
  [sneridagh]

- Remove Event, Link and Collection type.
  [sneridagh]

1.0.0a8 (2019-02-17)
--------------------

- Updated front page.
  [sneridagh]


1.0.0a7 (2019-02-16)
--------------------

- Remove from the main nav non folderish items.
  [sneridagh]


1.0.0a6 (2019-02-15)
--------------------

- Enable editing on the Plone Site with Volto Editor :)
  [sneridagh]


1.0.0a5 (2019-02-12)
--------------------

- Remove CORS, since we put it on the buildout
  [sneridagh]


1.0.0a4 (2019-02-12)
--------------------

- Remove Homepage content type
- Add default tiles to portal the proper way
- Add title
- Add CORS default config
  [sneridagh]


1.0.0a3 (2019-02-12)
--------------------

- Fix it as json loads spects strings...
  [sneridagh]

1.0.0a2 (2019-02-12)
--------------------

- Fix not set value on site serializer.
  [sneridagh]


1.0.0a1 (2019-02-12)
--------------------

- Initial release.
  [kitconcept]


src/kitconcept/voltodemo/setuphandlers.py
# -*- coding: utf-8 -*-
from plone import api
from Products.CMFPlone.interfaces import INonInstallable
from zope.interface import implementer

import json


@implementer(INonInstallable)
class HiddenProfiles(object):

    def getNonInstallableProfiles(self):
        """Hide uninstall profile from site-creation and quickinstaller"""
        return [
            'kitconcept.voltodemo:uninstall',
        ]


def post_install(context):
    """Post install script"""
    # portal = api.portal.get()

    create_default_homepage()


def uninstall(context):
    """Uninstall script"""
    # Do something at the end of the uninstallation of this package.


def add_catalog_indexes(context, wanted=None):
    """Method to add our wanted indexes to the portal_catalog.
    """
    catalog = api.portal.get_tool('portal_catalog')
    indexes = catalog.indexes()
    indexables = []
    for name, meta_type in wanted:
        if name not in indexes:
            catalog.addIndex(name, meta_type)
            indexables.append(name)
    if len(indexables) > 0:
        catalog.manage_reindexIndex(ids=indexables)


def create_default_homepage():
    portal = api.portal.get()

    blocks = {
      "0358abe2-b4f1-463d-a279-a63ea80daf19": {
          "@type": "description"
      },
      "07c273fc-8bfc-4e7d-a327-d513e5a945bb": {
          "@type": "title"
      },
      "2dfe8e4c-5bf6-43f1-93e1-6c320ede7226": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [],
                      "inlineStyleRanges": [
                          {
                              "length": 10,
                              "offset": 0,
                              "style": "BOLD"
                          }
                      ],
                      "key": "6470b",
                      "text": "Disclaimer: This instance is reset every night, so all changes will be lost afterwards.",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {}
          }
      },
      "3c881f51-f75b-4959-834a-6e1d5edc32ae": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [],
                      "inlineStyleRanges": [
                          {
                              "length": 5,
                              "offset": 6,
                              "style": "BOLD"
                          }
                      ],
                      "key": "ekn3l",
                      "text": "user: admin",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {}
          }
      },
      "5e1c30b1-ec6c-4dc0-9483-9768c3c416e4": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [
                          {
                              "key": 0,
                              "length": 5,
                              "offset": 0
                          },
                          {
                              "key": 1,
                              "length": 8,
                              "offset": 452
                          }
                      ],
                      "inlineStyleRanges": [],
                      "key": "behki",
                      "text": "Plone is a CMS built on Python with over 17 years of experience. Plone has very interesting features that appeal to developers and users alike, such as customizable content types, hierarchical URL object traversing and a sophisticated content workflow powered by a granular permissions model. This allows you to build anything from simple websites to enterprise-grade intranets. Volto exposes all these features and communicates with Plone via its mature REST API. Volto can be esily themed and is highly customizable.",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {
                  "0": {
                      "data": {
                          "href": "https://plone.org",
                          "rel": "nofollow",
                          "url": "https://plone.org/"
                      },
                      "mutability": "MUTABLE",
                      "type": "LINK"
                  },
                  "1": {
                      "data": {
                          "href": "https://github.com/plone/plone.restapi",
                          "url": "https://github.com/plone/plone.restapi"
                      },
                      "mutability": "MUTABLE",
                      "type": "LINK"
                  }
              }
          }
      },
      "61cc1bc0-d4f5-4e2b-9152-79512045a4dd": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [],
                      "inlineStyleRanges": [],
                      "key": "9qsa4",
                      "text": "Demo",
                      "type": "header-two"
                  }
              ],
              "entityMap": {}
          }
      },
      "874049e7-629e-489a-b46c-1adf35ad40ee": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [],
                      "inlineStyleRanges": [],
                      "key": "9pnjr",
                      "text": "Happy hacking!",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {}
          }
      },
      "942b6530-2407-420f-9c24-597adda6b2ce": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [
                          {
                              "key": 0,
                              "length": 36,
                              "offset": 39
                          }
                      ],
                      "inlineStyleRanges": [],
                      "key": "6a248",
                      "text": "Last but not least, it also supports a Volto Nodejs-based backend reference API implementation that demos how other systems could also use Volto to display and create content through it.",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {
                  "0": {
                      "data": {
                          "href": "https://github.com/plone/volto-reference-backend",
                          "url": "https://github.com/plone/volto-reference-backend"
                      },
                      "mutability": "MUTABLE",
                      "type": "LINK"
                  }
              }
          }
      },
      "9a976b8e-72ba-468a-bea8-b37a31bb386b": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [],
                      "inlineStyleRanges": [
                          {
                              "length": 12,
                              "offset": 51,
                              "style": "BOLD"
                          }
                      ],
                      "key": "94arl",
                      "text": "You can log in and use it as admin user using these credentials:",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {}
          }
      },
      "b3717238-448f-406e-b06f-57a9715c3326": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [
                          {
                              "key": 0,
                              "length": 5,
                              "offset": 0
                          }
                      ],
                      "inlineStyleRanges": [],
                      "key": "1bnna",
                      "text": "Volto is a React-based frontend for content management systems, currently supporting three backend implementations: Plone, Guillotina and a NodeJS reference implementation.",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {
                  "0": {
                      "data": {
                          "href": "https://github.com/plone/volto",
                          "url": "https://github.com/plone/volto"
                      },
                      "mutability": "MUTABLE",
                      "type": "LINK"
                  }
              }
          }
      },
      "c049ff8b-3e5a-4cfb-bca6-e4a6cca9be28": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [],
                      "inlineStyleRanges": [],
                      "key": "55n44",
                      "text": "You can use this site to test Volto. It runs the latest released Volto version using Plone 5.2 Backend running on Python 3.",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {}
          }
      },
      "c91f0fe9-f2e9-4a17-84a5-8e4f2678ed3c": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [],
                      "inlineStyleRanges": [
                          {
                              "length": 5,
                              "offset": 10,
                              "style": "BOLD"
                          }
                      ],
                      "key": "buncq",
                      "text": "password: admin",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {}
          }
      },
      "e0ca2fbc-7800-4b9b-afe5-8e42af9f5dd6": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [],
                      "inlineStyleRanges": [],
                      "key": "f0prj",
                      "text": "2019 - Volto Team - Plone Foundation",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {}
          }
      },
      "effbdcdc-253c-41a7-841e-5edb3b56ce32": {
          "@type": "text",
          "text": {
              "blocks": [
                  {
                      "data": {},
                      "depth": 0,
                      "entityRanges": [
                          {
                              "key": 0,
                              "length": 10,
                              "offset": 36
                          }
                      ],
                      "inlineStyleRanges": [],
                      "key": "68rve",
                      "text": "Volto also supports other APIs like Guillotina, a Python resource management system, inspired by Plone and using the same basic concepts like traversal, content types and permissions model.",
                      "type": "unstyled"
                  }
              ],
              "entityMap": {
                  "0": {
                      "data": {
                          "href": "https://guillotina.io/",
                          "rel": "nofollow",
                          "url": "https://guillotina.io/"
                      },
                      "mutability": "MUTABLE",
                      "type": "LINK"
                  }
              }
          }
      }
    }

    blocks_layout = {
      "items": [
          "07c273fc-8bfc-4e7d-a327-d513e5a945bb",
          "0358abe2-b4f1-463d-a279-a63ea80daf19",
          "b3717238-448f-406e-b06f-57a9715c3326",
          "5e1c30b1-ec6c-4dc0-9483-9768c3c416e4",
          "effbdcdc-253c-41a7-841e-5edb3b56ce32",
          "942b6530-2407-420f-9c24-597adda6b2ce",
          "61cc1bc0-d4f5-4e2b-9152-79512045a4dd",
          "c049ff8b-3e5a-4cfb-bca6-e4a6cca9be28",
          "9a976b8e-72ba-468a-bea8-b37a31bb386b",
          "3c881f51-f75b-4959-834a-6e1d5edc32ae",
          "c91f0fe9-f2e9-4a17-84a5-8e4f2678ed3c",
          "2dfe8e4c-5bf6-43f1-93e1-6c320ede7226",
          "874049e7-629e-489a-b46c-1adf35ad40ee",
          "e0ca2fbc-7800-4b9b-afe5-8e42af9f5dd6"
      ]
    }

    if not getattr(portal, 'blocks', False):
        portal.manage_addProperty('blocks', json.dumps(blocks), 'string')

    if not getattr(portal, 'blocks_layout', False):
        portal.manage_addProperty('blocks_layout', json.dumps(blocks_layout), 'string') # noqa

    portal.setTitle('Welcome to Volto!')
    portal.setDescription('The React powered content management system')
